testsuite/brownian_thermostat.py

```diff
@@ -0,0 +1,413 @@
+#        # Integral of vacf via Green-Kubo
+#        #D= int_0^infty <v(t_0)v(t_0+t)> dt     (o 1/3, since we work componentwise)
+#        acf=c.result()
+#        #Integrate w. trapez rule
+#        for coord in 2,3,4:
+#            I=np.trapz(acf[:,coord],acf[:,0])
+#            ratio = I/(kT/gamma[coord-2])
+#            self.assertAlmostEqual(ratio,1.,delta=0.07)
+        
+
+#    TODO: to return this after an implementation of the fine inertial BD
+#    def test_00__friction_trans(self):
+#        """Tests the translational friction-only part of the thermostat."""
+#
+#        
+#        s=self.s
+#        # Translation
+#        gamma_t_i=2
+#        gamma_t_a=0.5,2,1.5
+#        v0=5.
+#
+#        s.time_step=0.0005
+#        s.part.clear()
+#        s.part.add(pos=(0,0,0),v=(v0,v0,v0))
+#        if espressomd.has_features("MASS"):
+#            s.part[0].mass=3
+#        if espressomd.has_features("PARTICLE_ANISOTROPY"):
+#            s.thermostat.set_brownian(kT=0,gamma=gamma_t_a)
+#        else:
+#            s.thermostat.set_brownian(kT=0,gamma=gamma_t_i)
+#
+#        s.time=0
+#        for i in range(100):
+#            s.integrator.run(10)
+#            for j in range(3):
+#                if espressomd.has_features("PARTICLE_ANISOTROPY"):
+#                    self.assertAlmostEqual(s.part[0].v[j],v0*np.exp(-gamma_t_a[j]/s.part[0].mass*s.time),places=2)
+#                else:
+#                    self.assertAlmostEqual(s.part[0].v[j],v0*np.exp(-gamma_t_i/s.part[0].mass*s.time),places=2)
+
+#    TODO: to return this after an implementation of the fine inertial BD
+#    @ut.skipIf(not espressomd.has_features("ROTATION"), "Skipped for lack of ROTATION" )
+#    def test_00__friction_rot(self):
+#        """Tests the rotational friction-only part of the thermostat."""
+#
+#        
+#        s=self.s
+#        # Translation
+#        gamma_t_i=2
+#        gamma_t_a=0.5,2,1.5
+#        gamma_r_i=3
+#        gamma_r_a=1.5,0.7,1.2
+#        o0=5.
+#
+#        s.time_step=0.0005
+#        s.part.clear()
+#        s.part.add(pos=(0,0,0),omega_body=(o0,o0,o0),rotation=(1,1,1))
+#        if espressomd.has_features("ROTATIONAL_INERTIA"):
+#            s.part[0].rinertia=2,2,2
+#        if espressomd.has_features("PARTICLE_ANISOTROPY"):
+#            s.thermostat.set_brownian(kT=0,gamma=gamma_t_a,gamma_rotation=gamma_r_a)
+#        else:
+#            s.thermostat.set_brownian(kT=0,gamma=gamma_t_i,gamma_rotation=gamma_r_i)
+#
+#        s.time=0
+#        for i in range(100):
+#            s.integrator.run(10)
+#            for j in range(3):
+#                if espressomd.has_features("PARTICLE_ANISOTROPY"):
+#                    self.assertAlmostEqual(s.part[0].omega_body[j],o0*np.exp(-gamma_r_a[j]/s.part[0].rinertia[j]*s.time),places=2)
+#                else:
+#                    self.assertAlmostEqual(s.part[0].omega_body[j],o0*np.exp(-gamma_r_i/s.part[0].rinertia[j]*s.time),places=2)
+
+
+
+
+if __name__ == "__main__":
+    ut.main()
```

Comment:
You have mixed indentation (4 spaces and 3 spaces).  Please make it 4 spaces throughout.

Reply:
Agree, I've fixed this after the pep8 check.